lib/usd/utils/tests/test_MayaTransformAPI.cpp

```diff
@@ -0,0 +1,1196 @@
+
+  {
+    UsdStageRefPtr stage = UsdStage::CreateInMemory();
+    UsdGeomXform xform = UsdGeomXform::Define(stage, SdfPath("/maya_xform"));
+
+    MayaUsdUtils::MayaTransformAPI api(xform.GetPrim());
+    EXPECT_TRUE(api);
+
+    api.rotate(GfVec3f(1, 2, 3), MayaUsdUtils::RotationOrder::kYZX);
+    auto value = api.rotate();
+    EXPECT_NEAR(value[0], 1.0f, 0.1f);
+    EXPECT_NEAR(value[1], 2.0f, 0.1f);
+    EXPECT_NEAR(value[2], 3.0f, 0.1f);
+    EXPECT_EQ(MayaUsdUtils::RotationOrder::kYZX, api.rotateOrder());
+
+    bool resetsXformStack;
+    std::vector<UsdGeomXformOp> ops = xform.GetOrderedXformOps(&resetsXformStack);
+
+    ASSERT_EQ(1U, ops.size());
+    EXPECT_EQ(UsdGeomXformOp::TypeRotateYZX, ops[0].GetOpType());
+    EXPECT_EQ(UsdGeomXformOp::PrecisionFloat, ops[0].GetPrecision());
+  }
+
+  {
+    UsdStageRefPtr stage = UsdStage::CreateInMemory();
+    UsdGeomXform xform = UsdGeomXform::Define(stage, SdfPath("/maya_xform"));
+
+    MayaUsdUtils::MayaTransformAPI api(xform.GetPrim());
+    EXPECT_TRUE(api);
+
+    api.rotate(GfVec3f(1, 2, 3), MayaUsdUtils::RotationOrder::kZXY);
+    auto value = api.rotate();
+    EXPECT_NEAR(value[0], 1.0f, 0.1f);
+    EXPECT_NEAR(value[1], 2.0f, 0.1f);
+    EXPECT_NEAR(value[2], 3.0f, 0.1f);
+    EXPECT_EQ(MayaUsdUtils::RotationOrder::kZXY, api.rotateOrder());
+
+    bool resetsXformStack;
+    std::vector<UsdGeomXformOp> ops = xform.GetOrderedXformOps(&resetsXformStack);
+
+    ASSERT_EQ(1U, ops.size());
+    EXPECT_EQ(UsdGeomXformOp::TypeRotateZXY, ops[0].GetOpType());
+    EXPECT_EQ(UsdGeomXformOp::PrecisionFloat, ops[0].GetPrecision());
+  }
+
+  {
+    UsdStageRefPtr stage = UsdStage::CreateInMemory();
+    UsdGeomXform xform = UsdGeomXform::Define(stage, SdfPath("/maya_xform"));
+
+    MayaUsdUtils::MayaTransformAPI api(xform.GetPrim());
+    EXPECT_TRUE(api);
+
+    api.rotate(GfVec3f(1, 2, 3), MayaUsdUtils::RotationOrder::kZYX);
+    auto value = api.rotate();
+    EXPECT_NEAR(value[0], 1.0f, 0.1f);
+    EXPECT_NEAR(value[1], 2.0f, 0.1f);
+    EXPECT_NEAR(value[2], 3.0f, 0.1f);
+    EXPECT_EQ(MayaUsdUtils::RotationOrder::kZYX, api.rotateOrder());
+
+    bool resetsXformStack;
+    std::vector<UsdGeomXformOp> ops = xform.GetOrderedXformOps(&resetsXformStack);
+
+    ASSERT_EQ(1U, ops.size());
+    EXPECT_EQ(UsdGeomXformOp::TypeRotateZYX, ops[0].GetOpType());
+    EXPECT_EQ(UsdGeomXformOp::PrecisionFloat, ops[0].GetPrecision());
+  }
+
+  {
+    UsdStageRefPtr stage = UsdStage::CreateInMemory();
+    UsdGeomXform xform = UsdGeomXform::Define(stage, SdfPath("/maya_xform"));
+
+    MayaUsdUtils::MayaTransformAPI api(xform.GetPrim());
+    EXPECT_TRUE(api);
+
+    api.inheritsTransform(true);
+    EXPECT_EQ(true, api.inheritsTransform());
+
+    api.inheritsTransform(false);
+    EXPECT_EQ(false, api.inheritsTransform());
+  }
+}
+
+const char* getDataPath(const char* const str)
+{
+  static char buffer[512];
+  sprintf(buffer, "%s/%s.usda", AL_EXTRAS_TEST_DATA, str);
+  return buffer;
+}
+
+/*
```

Comment:
Commented out code?

Reply:
Yeah, that was the MEL used to generate the unit tests. Thought it made sense to leave it there for someone in the future, but can remove.